### PR TITLE
PROD-778 #2696-fix-Title showing wrong if Rank Math SEO activated

### LINKED ISSUE #2696: Title showing wrong if Rank Math SEO activated
**Describe the bug**
Activate the Rank Math SEO plugin and open the following pages User Timeline, User Connection, User Groups, then it will only show the username and site title.

**To Reproduce**
1. Go to website > user profile dropdown
2. Click on Timeline
3. Title showing wrong

**Expected behavior**
The title should display according to the page.

**Screenshots**
https://vimeo.com/580690958/b81bfb805a

**Environment**
Which site did you replicate this issue on?
1) Customer: https://wordpressninja.net/members/
2) My Dev site: https://dev-milan.flywheelsites.com/
3) My local machine 

**Support ticket links**
https://secure.helpscout.net/conversation/1575649197/154268/

**Jira issue** : [PROD-778]

[PROD-778]: https://buddyboss.atlassian.net/browse/PROD-778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by meetmakadiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2696  .

### How to test the changes in this Pull Request:

1. Activate the Rank Math SEO plugin.
2. Go to the frontend user connection, feed, groups tab, and check the title.

### Proof Screenshots or Video

https://www.loom.com/share/025582bf6454454792932454b54970ae

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — DISMISSED** [review]

**Comment by basil-buddyboss** [comment]
Retested in the QA ongoing branch.
Status: Not Fixed

Proof Screenshots or Video:
https://watch.screencastify.com/v/GdjzTYNT1YthCVKZJTKB

File: src/bp-core/compatibility/bp-rankmath-plugin-helpers.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ class BP_RankMath_Title implements IPaper {
 	 * @return string
 	 */
 	public function title() {
-		if ( bp_is_user() ) {
+		if ( bp_is_user() && bp_is_current_component('xprofile') ) {
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-rankmath-plugin-helpers.php:26]:
```suggestion
		if ( bp_is_user() && bp_is_current_component( 'xprofile' ) ) {
```

File: src/bp-core/compatibility/bp-rankmath-plugin-helpers.php [hunk 2]
```diff
@@ -91,10 +91,7 @@ function bp_helper_rankmath_group_page_support( $title ) {
 		return;
 	}
 
-	if (
-		bp_is_active( 'groups' ) && ! empty( buddypress()->groups->current_group )
-		|| bp_is_user()
-	) {
+	if ( bp_is_active( 'groups' ) && ! empty( buddypress()->groups->current_group ) || bp_is_current_component('groups') ) {
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-rankmath-plugin-helpers.php:94]:
```suggestion
	if ( bp_is_active( 'groups' ) && ( ! empty( buddypress()->groups->current_group ) || bp_is_current_component( 'groups' ) ) ) {
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Review by KartikSuthar — APPROVED** [review]

File: src/bp-core/compatibility/bp-rankmath-plugin-helpers.php [hunk 3]
```diff
@@ -23,13 +23,23 @@ class BP_RankMath_Title implements IPaper {
...
+			$title  = isset( buddypress()->groups->current_group->name ) ? buddypress()->groups->current_group->name : esc_html__( 'Social Group', 'buddyboss' );
+			if( 'admin' === $action ) {
+				$action = esc_html__( 'Manage', 'buddyboss' );
+			}
+			if( 'my-groups' === $action ) {
+				$action = esc_html__( 'Groups', 'buddyboss' );
+			}
+			if( isset( $action ) && !empty( $action ) ) {
+				$title = ucfirst( $action ) . ' - ' . $title;
+			}
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-rankmath-plugin-helpers.php:42]: @meetmakadiya  follow PHPCS here.

Comment (meetmakadiya) [reply src/bp-core/compatibility/bp-rankmath-plugin-helpers.php]: @KartikSuthar fixed the phpcs.

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**